datavault-common/src/main/java/org/datavaultplatform/common/model/dao/RoleAssignmentDaoImpl.java

```diff
@@ -20,6 +22,55 @@ public void setSessionFactory(SessionFactory sessionFactory) {
         this.sessionFactory = sessionFactory;
     }
 
+    @Override
+    public void synchroniseDataOwners() {
```

Comment:
Any reason this isn't just a one off migration? With large enough datasets this will eventually get to the point of not being able to scale with the amount of data it needs to read and write from the database server.

Reply:
No particular reason - I needed to be able to hold the ID of the Data Owner role in a variable for the migration script which I figured wasn't going to be available in MySQL as its PL/SQL equivalent is really limited. But it looks like it does have variables so I'll see if I can get this working via a script instead

Reply:
I don't think it matters if it's SQL or Java code that does the migration, but it should probably only be a one off.

I think an easy solution might be introducing a toggle flag (`datavault.shouldMigrateDataOwners` or something) that you can switch off after letting it run once. WDYT?

Reply:
I'd be in favour of that if this was our product because we could remove the dead code once we know it's dead, but I don't like using feature toggles as permanent ways of turning off dead code. I think an SQL script would be the best thing here

Reply:
I agree an SQL script is best (although a bit more work, hence the suggestion of a flag), but I wouldn't preclude a code solution for the reason given. Their team is (and should be) perfectly capable of taking care of post-migration clean-up.